app/src/main/java/com/github/lookupgroup27/lookup/model/map/renderables/Planet.kt

```diff
@@ -0,0 +1,178 @@
+package com.github.lookupgroup27.lookup.model.map.renderables
+
+import android.content.Context
+import android.opengl.GLES20
+import android.opengl.Matrix
+import com.github.lookupgroup27.lookup.model.map.renderables.utils.Sphere.generateSphericalGeometry
+import com.github.lookupgroup27.lookup.model.map.skybox.buffers.ColorBuffer
+import com.github.lookupgroup27.lookup.model.map.skybox.buffers.IndexBuffer
+import com.github.lookupgroup27.lookup.model.map.skybox.buffers.TextureBuffer
+import com.github.lookupgroup27.lookup.model.map.skybox.buffers.VertexBuffer
+import com.github.lookupgroup27.lookup.util.opengl.ShaderProgram
+import com.github.lookupgroup27.lookup.util.opengl.TextureManager
+
+/**
+ * Represents a 3D planet rendered in an OpenGL environment. This class handles the geometry,
+ * textures, and shaders required to render a spherical planet with customizable appearance and
+ * position.
+ *
+ * Features include:
+ * - Geometry generation using sphere tessellation.
+ * - Support for custom textures, colors, and scaling.
+ * - Integration with OpenGL shaders for rendering.
+ *
+ * @property context The Android context used for resource access.
+ * @property name The name of the planet (e.g., "Earth"). Defaults to "Planet".
+ * @property position The planet's position in 3D space, represented as a float array [x, y, z].
+ * @property textureId The resource ID of the texture applied to the planet's surface.
+ * @property numBands The number of latitude bands used for tessellating the sphere. Higher values
+ *   create smoother spheres.
+ * @property stepsPerBand The number of longitude steps per latitude band. Higher values improve
+ *   rendering fidelity.
+ */
+class Planet(
+    private val context: Context,
+    private val name: String = "Planet",
+    private val position: FloatArray = floatArrayOf(0.0f, 0.0f, 0.0f),
+    private val textureId: Int,
+    private val numBands: Int = DEFAULT_NUM_BANDS,
+    private val stepsPerBand: Int = DEFAULT_STEPS_PER_BAND
+) {
```

Comment:
To maintain consistency with our structure, please have this class inherit from the `Object` class. Feel free to modify the `Object` class if you think it can be improved, but ensure that it aligns with the current architecture. Let me know if you need any input!